File: archive/c/clojure/baklava.clj
(ns baklava
  (:gen-class)
  (:require [clojure.string :refer [join]])
)

(defn repeat-string [n, s]
  (join (repeat n s))
)

(defn baklava-line [n]
  (def num-spaces (abs (- n 10)))
  (def num-stars (- 21 (* 2 num-spaces)))
  (str (repeat-string num-spaces " ") (repeat-string num-stars "*"))
)

(defn baklava [n]
  (join "\n" (map baklava-line (range 0 n)))
)

(println (baklava 21))

File: archive/s/scala/Baklava.scala
object Baklava:
  def generate(size: Int): Seq[String] =
    for i <- -size to size yield
      val spaces = i.abs
      val stars = (size * 2 + 1) - 2 * spaces
      " " * spaces + "*" * stars

  def main(args: Array[String]): Unit =
    val size =
      if args.nonEmpty then args(0).toInt
      else 10

    generate(size).foreach(println)